internal/provider/resource_api_key.go

```diff
@@ -0,0 +1,141 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	neon "github.com/kislerdm/neon-sdk-go"
+)
+
+func resourceAPIKey() *schema.Resource {
+	return &schema.Resource{
+		Description:   "A key to access the Neon API.",
+		SchemaVersion: 7,
```

Comment:
**(suggestion, blocking)** I'd suggest to start with 1 because it's the new resource so we could version its schema separately.

```suggestion
      SchemaVersion: 1,
```